src/app/modules/tree-viewer/tree-viewer.component.spec.ts

```diff
@@ -0,0 +1,207 @@
+import { async, ComponentFixture, TestBed, fakeAsync, tick } from '@angular/core/testing';
+
+import { TreeViewerComponent } from './tree-viewer.component';
+import { TreeNode } from './tree-node';
+import { TreeViewerConfig } from './tree-viewer-config';
+import { By } from '@angular/platform-browser';
+import { DebugElement } from '@angular/core';
+
+describe('TreeViewerComponent', () => {
+  let component: TreeViewerComponent;
+  let fixture: ComponentFixture<TreeViewerComponent>;
+
+  const singleEmptyTreeNode: () => TreeNode = () => ({
+    name: 'tree node',
+    children: [],
+    leafStyle: 'fa-file'
+  });
+
+  const treeNodeWithSubNodes: () => TreeNode = () => ({
+    name: 'parent node',
+    collapsedStyle: 'fa-chevron-right',
+    expandedStyle: 'fa-chevron-down',
+    leafStyle: 'fa-folder',
+    expanded: false,
+    children: [
+      { name: 'child node 1', children: [] },
+      { name: 'child node 2', children: [] },
+      { name: 'child node 3', children: [] }
+    ]
+  });
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [ TreeViewerComponent ]
+    })
+    .compileComponents();
+  }));
+
+  function getItemKey(): DebugElement {
```

Comment:
getRootTreeViewItemKey (is a bit long, but is otherwise a bit misleading in the presence of multiple tree-view-item-key 's, isn't it)